src/main/java/com/google/sps/servlets/LiveCourseData.java

```diff
@@ -0,0 +1,94 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/term-live")
+public class LiveCourseData extends HttpServlet {
+  private DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    getTerm(db, request);
+    response.setContentType("text/html; charset=UTF-8");
+    response.setCharacterEncoding("UTF-8");
+    response.sendRedirect("/AddSchoolData.html");
+  }
+
+  public Entity getTerm(DatastoreService db, HttpServletRequest request) {
+    String schoolName = request.getParameter("school-name");
+    String courseName = request.getParameter("course-name");
+    String termName = request.getParameter("term");
+    String profName = request.getParameter("professor-name");
+    Long units = Long.parseLong(request.getParameter("units"));
+
+    Entity foundTerm = findTerm(db, schoolName, courseName, termName, units, profName);
+    return foundTerm;
+  }
+
+  public List<Long> getTermData(DatastoreService db, Entity term, String property) {
```

Comment:
Data is not a helpful name usually. It looks like this is just ratings for now so maybe call it that? Easy enough to change later.

When you do change, then ideally we'd create a record of some sort to denote what kind of data we're passing around.

Reply:
Ah I see from the test you're getting metadata from the ratings, not the ratings themselves. 
In that case, something like getTermDataFromRatngs or something could be helpful. Up to you.